enhancements/baremetal/baremetal-config-raid-and-bios.md

```diff
@@ -0,0 +1,149 @@
+3. [Terraform-provider-ironic](https://github.com/openshift-metal3/terraform-provider-ironic) gets the
+configuration of RAID and BIOS for control plane hosts and then passes it to Ironic([Baremetal Operator(BMO)](https://github.com/metal3-io/baremetal-operator)
+has supported the process of RAID and BIOS for worker nodes).
+
+### User Stories
+
+With the addition of this feature, the users can configure RAID and BIOS in IPI deployments.
+
+### Implementation Details/Notes/Constraints
+
+#### Fill two fields
+
+The user runs `create manifests` and then fill *raid* and *firmware* fields in the **BMH** created by the installer.
+
+```yaml
+spec:
+  bmc:
+    address: <bmc>://<ip-address>
+    credentialsName: worker-0-bmc-secret
+  bootMACAddress: mac-address
+  raid: raid-config
+  firmware: bios-config
+  ...
+```
+
+#### Process the fields in installer
+
+For control plane hosts, transform the *raid* and *firmware* fields into string respectively and then write them into the
+***terraform.baremetal.auto.tfvars.json*** file as the values of *raid_config* and *bios_settings* respectively.
+
+#### Process the fields in terraform-provider-ironic
+
+Add *raid_config* and *bios_settings* fields to terraform-provider-ironic API, transform them back to struct format, and then call the
+methods in **BMO** to process the two fields to perform [manual cleaning](https://docs.openstack.org/ironic/latest/admin/cleaning.html#manual-cleaning).
+
+#### Notes
+
+When the IPI deployments completed, the **BMH** resources will still get persisted in the cluster, though control plane hosts have
+RAID and BIOS field in their **BMH**, but the *externallyProvisioned* field's value is true, so the RAID and BIOS configurations are
+invalid.
+
+### Risks and Mitigations
+
+N/A
+
+## Design Details
+
+### Test Plan
+
+- Unit tests for determining the configuration of RAID and BIOS passed to Ironic meeting expectations.
+- e2e tests for determining the configuration of RAID and BIOS configured during the IPI deployments.
+
+### Graduation Criteria
+
+N/A
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+Older versions of the installer will ignore the *raid* and *firmware* fields.
+
+### Version Skew Strategy
+
+N/A
+
+## Implementation History
+
+- [Implementation on openshift/installer](https://github.com/hs0210/installer/tree/ipi-support-raid-bios)
+- [Implementation on openshift-metal3/terraform-provider-ironic](https://github.com/hs0210/terraform-provider-ironic/tree/ipi-support-raid-bios)
+
+## Drawbacks
+
+This will increase the number of steps in IPI deployments and take longer.
+
+## Alternatives
+
+Add RAID and BIOS configuration by modifying the manifest files **~/clusterconfigs/openshift/99_openshift-cluster-api_hosts- \*.yaml** generated by executing `openshift-baremetal-install --dir ~/clusterconfigs create manifest`.
```

Comment:
This alternative is the same as the proposal above - we could perhaps make it clearer if we say something like:

```
It is already possible to modify the manifest files generated by the installer, but currently any changes to bios or RAID config will only have any affect for workers hosts.  It is desirable to enable configuration of both master and worker hosts via the same method.
```

Reply:
Thanks, done.